cmd/juju/application/store/charmadapter.go

```diff
@@ -89,41 +80,14 @@ func (c *CharmAdaptor) ResolveCharm(url *charm.URL, preferredOrigin commoncharm.
 		err = resolved[0].Error
 	}
 
-	// In order to maintain backwards compatibility with the charmstore, we need
-	// to correctly handle charmhub vs charmstore failures.
-	// The following attempts to correctly fallback if the url is a charmhub
-	// charm, otherwise fallback to how the old client worked.
-	if charm.CharmHub.Matches(url.Schema) {
-		if errors.IsNotSupported(err) {
-			return nil, commoncharm.Origin{}, nil, errors.NewNotSupported(nil, "charmhub charms are not supported by the current controller; if you wish to use charmhub consider upgrading your controller to 2.9+.")
-		}
-		return nil, commoncharm.Origin{}, nil, errors.Trace(err)
-	}
-	return c.resolveCharmFallback(url, preferredOrigin)
-}
-
-func (c *CharmAdaptor) resolveCharmFallback(url *charm.URL, preferredOrigin commoncharm.Origin) (*charm.URL, commoncharm.Origin, []string, error) {
-	charmRepo, err := c.charmStoreRepoFunc()
-	if err != nil {
-		return nil, commoncharm.Origin{}, nil, errors.Trace(err)
-	}
-
-	resultURL, channel, supportedSeries, err := charmRepo.ResolveWithPreferredChannel(url, csparams.Channel(preferredOrigin.Risk))
-	if err != nil {
-		// Ideally the would be restored before now, however the
-		// callee is in the core package, let's not add dependecies there.
-		return nil, commoncharm.Origin{}, nil, errors.Trace(apiservererrors.RestoreError(err))
+	if errors.Is(err, errors.NotSupported) {
+		return nil, commoncharm.Origin{}, nil, errors.NewNotSupported(nil, "charmhub charms are not supported by the current controller; if you wish to use charmhub consider upgrading your controller to 2.9+.")
 	}
-	origin := preferredOrigin
-	origin.Risk = string(channel)
-	if resultURL.Series != "" && len(supportedSeries) == 0 {
-		supportedSeries = []string{resultURL.Series}
-	}
-	return resultURL, origin, supportedSeries, nil
+	return nil, commoncharm.Origin{}, nil, errors.Trace(err)
 }
 
-// ResolveBundleURL tries to interpret maybeBundle as a CharmStore
-// or CharmHub bundle. If it turns out to be a bundle, the resolved
+// ResolveBundleURL tries to interpret maybeBundle as a CharmHub
```

Comment:
I think it's "Charmhub", only one capital letter. I usually do the same thing, incorrectly.